spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/core/cloudfoundry/GcpCloudFoundryEnvironmentPostProcessor.java

```diff
@@ -0,0 +1,129 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.core.cloudfoundry;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.cloud.CloudPlatform;
+import org.springframework.boot.context.config.ConfigFileApplicationListener;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.boot.json.JsonParser;
+import org.springframework.boot.json.JsonParserFactory;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertiesPropertySource;
+
+/**
+ * @author João André Martins
+ */
+public class GcpCloudFoundryEnvironmentPostProcessor
+		implements EnvironmentPostProcessor, Ordered {
+
+	public static final String VCAP_SERVICES_ENVVAR = "VCAP_SERVICES";
+
+	private static final Log LOGGER =
+			LogFactory.getLog(GcpCloudFoundryEnvironmentPostProcessor.class);
+
+	private static final String SPRING_CLOUD_GCP_PROPERTY_PREFIX = "spring.cloud.gcp.";
+
+	private JsonParser parser = JsonParserFactory.getJsonParser();
+
+	private int order = ConfigFileApplicationListener.DEFAULT_ORDER - 1;
+
+	private enum GcpCfService {
+
+		pubsub("google-pubsub",
+				ImmutableMap.of("ProjectId", "project-id",
+				"PrivateKeyData", "credentials.encodedKey",
+				"topic_name", "topic-name",
+				"subscription_name", "subscription-name")),
+		storage("google-storage",
+				ImmutableMap.of("ProjectId", "project-id",
+				"PrivateKeyData", "credentials.encodedKey",
+				"bucket_name", "bucket-name"));
+
+		private String cfServiceName;
+		private Map<String, String> cfPropNameToGcp;
+
+		GcpCfService(String cfServiceName,
+				Map<String, String> cfPropNameToGcp) {
+			this.cfServiceName = cfServiceName;
+			this.cfPropNameToGcp = cfPropNameToGcp;
+		}
+
+		public String getCfServiceName() {
+			return this.cfServiceName;
+		}
+
+		public Map<String, String> getCfPropNameToGcp() {
+			return this.cfPropNameToGcp;
+		}
+	}
+
+	@Override
+	public void postProcessEnvironment(ConfigurableEnvironment environment,
+			SpringApplication application) {
+		if (CloudPlatform.CLOUD_FOUNDRY.isActive(environment)) {
```

Comment:
You might remove that check (and just check for the presence of VCAPs) as it would ease unit testing of this feature. Practically that code is doing that but it might change in the future so doing it explicitly is better IMO